source/game/system/RaceManager.hh

```diff
@@ -1,19 +1,32 @@
 #pragma once
 
 #include "game/system/KPadController.hh"
+#include "game/system/map/MapdataCheckPoint.hh"
 
 #include <egg/math/Vector.hh>
 
+#include <vector>
+
 namespace System {
 
 class RaceManagerPlayer {
 public:
     RaceManagerPlayer();
     virtual ~RaceManagerPlayer() {}
 
+    void init();
+    void calc();
+
     [[nodiscard]] const KPad *inputs() const;
+    [[nodiscard]] u8 respawn() const;
 
 private:
+    MapdataCheckPoint *calcCheckpoint(u16 checkpointId, f32 distanceRatio);
+    bool areCheckpointsSubsequent(MapdataCheckPoint *checkpoint, u16 nextCheckpointId) const;
+
+    u16 m_checkpointId;
+    f32 m_checkpointFactor; ///< The proportion of a lap for the current checkpoint
+    u8 m_respawn;
```

Comment:
```suggestion
    u8 m_jugemId;
```
Given that we refer to this in the checkpoint as `jugemIndex`.

Reply:
Done. I was going to also rename the `respawn()` getter function but I see that it's actually unused, so I am removing that getter from the PR